test/test_parser.cpp

```diff
@@ -1840,35 +1842,35 @@ TEST(Parser_BacklinkCount)
     list_2->add(2);
 
     verify_query(test_context, items, "@links.@count == 0", 1);
-    verify_query(test_context, items, "@links.@count == 0 && item_id == 4", 1);
+    verify_query(test_context, items, "@links.@count == 0 && item_id == 20", 1);
     verify_query(test_context, items, "@links.@count == 1", 1);
-    verify_query(test_context, items, "@links.@count == 1 && item_id == 3", 1);
+    verify_query(test_context, items, "@links.@count == 1 && item_id == 14", 1);
     verify_query(test_context, items, "@links.@count == 5", 1);
-    verify_query(test_context, items, "@links.@count == 5 && item_id == 2", 1);
+    verify_query(test_context, items, "@links.@count == 5 && item_id == 12", 1);
     verify_query(test_context, items, "@links.@count == 3", 1);
-    verify_query(test_context, items, "@links.@count == 3 && item_id == 1", 1);
+    verify_query(test_context, items, "@links.@count == 3 && item_id == 2", 1);
     verify_query(test_context, items, "@links.@count == 13", 1);
-    verify_query(test_context, items, "@links.@count == 13 && item_id == 0", 1);
+    verify_query(test_context, items, "@links.@count == 13 && item_id == 5", 1);
 
     // @size is still a synonym to @count
     verify_query(test_context, items, "@links.@size == 0", 1);
-    verify_query(test_context, items, "@links.@size == 0 && item_id == 4", 1);
+    verify_query(test_context, items, "@links.@size == 0 && item_id == 20", 1);
 
     // backlink count through forward links
-    verify_query(test_context, t, "fav_item.@links.@count == 3 && fav_item.item_id == 1", 1);
-    verify_query(test_context, t, "fav_item.@links.@count == 13 && fav_item.item_id == 0", 1);
+    verify_query(test_context, t, "fav_item.@links.@count == 5 && fav_item.item_id == 12", 1);
+    verify_query(test_context, t, "fav_item.@links.@count == 13 && fav_item.item_id == 5", 1);
 
     // backlink count through backlinks first
-    verify_query(test_context, items, "@links.class_Items.self.@links.@count == 1 && item_id == 3", 1);
+    verify_query(test_context, items, "@links.class_Items.self.@links.@count == 1 && item_id == 14", 1);
     verify_query(test_context, items, "@links.class_Person.items.@links.@count == 0", 5);
 
     // backlink count through backlinks and forward links
-    verify_query(test_context, items, "@links.class_Person.fav_item.items.@links.@count == 1 && item_id == 3", 1);
-    verify_query(test_context, items, "@links.class_Person.fav_item.fav_item.@links.@count == 1 && item_id == 3", 1);
+    verify_query(test_context, items, "@links.class_Person.fav_item.items.@links.@count == 130 && item_id == 2", 1);
+    verify_query(test_context, items, "@links.class_Person.fav_item.fav_item.@links.@count == 3 && item_id == 2", 1);
```

Comment:
These two checks are not correct actually, as far as I understand we shouldn't sum the backlinks at each stage of the link, just return the backlink count at the destination table. I think it was correct before?